tasks.c

```diff
@@ -4905,7 +4905,11 @@ BaseType_t xTaskIncrementTick( void )
                         if( pxCurrentTCBs[ xCoreID ]->xPreemptionDisable == pdFALSE )
                     #endif
                     {
-                        if( ( xYieldRequiredForCore[ xCoreID ] != pdFALSE ) || ( xYieldPendings[ xCoreID ] != pdFALSE ) )
+                        #if ( configUSE_TIME_SLICING == 1 )
+                            if( ( xYieldRequiredForCore[ xCoreID ] != pdFALSE ) || ( xYieldPendings[ xCoreID ] != pdFALSE ) )
+                        #else
+                            if( xYieldPendings[ xCoreID ] != pdFALSE )
+                        #endif /* #if ( configUSE_TIME_SLICING == 1 ) */
```

Comment:
This looks fine to me now, but can't help but wonder if we should eliminate `xYieldRequiredForCore` altogether.  (Sorry for not seeing this earlier.)  The code above that sets `xYieldRequiredForCore[ xCoreID ]` would just set xYieldPendings[ xCoreID ] instead, leaving the original test-coverage marker in the "else" case.  Would you mind investigating that possibility?  Any concerns @aggarg?